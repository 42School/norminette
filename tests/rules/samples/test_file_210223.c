int	main(void)
{
	if (pthread_create(&philos[i].philo_status_thread, NULL, \
		check_status, &philos[i]))
		return (-1);
	printf("\nOh noes. I, nÂ°%i, no longer thinks, and therefore, is no more.\
	*dies in philosopher*\n", this->uid);
}

int	main(void)
{
	if ((-(cy->height * 0.5)) < z[0] && z[0] < ((cy->height * 0.5)))
		return ;
}
