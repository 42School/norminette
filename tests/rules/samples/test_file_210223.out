[36mtest_file_210223.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 1":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 2":
		<LBRACE> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsControlStatement[0m In "Function" from "GlobalScope" line 3":
		<TAB> <IF> <SPACE> <LPARENTHESIS> <IDENTIFIER=pthread_create> <LPARENTHESIS> <BWISE_AND> <IDENTIFIER=philos> <LBRACKET> <IDENTIFIER=i> <RBRACKET> <DOT> <IDENTIFIER=philo_status_thread> <COMMA> <SPACE> <NULL> <COMMA> <SPACE> <ESCAPED_NEWLINE> <TAB> <TAB> <IDENTIFIER=check_status> <COMMA> <SPACE> <BWISE_AND> <IDENTIFIER=philos> <LBRACKET> <IDENTIFIER=i> <RBRACKET> <RPARENTHESIS> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsExpressionStatement[0m In "ControlStructure" from "Function" line 5":
		<TAB> <TAB> <RETURN> <SPACE> <LPARENTHESIS> <MINUS> <CONSTANT=1> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsFunctionCall[0m In "Function" from "GlobalScope" line 6":
		<TAB> <IDENTIFIER=printf> <LPARENTHESIS> <STRING="\nOh noes. I, nÂ°%i, no longer thinks, and therefore, is no more.\
	*dies in philosopher*\n"> <COMMA> <SPACE> <IDENTIFIER=this> <PTR> <IDENTIFIER=uid> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 8":
		<RBRACE> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 9":
		<NEWLINE>
[36mtest_file_210223.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 10":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 11":
		<LBRACE> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsControlStatement[0m In "Function" from "GlobalScope" line 12":
		<TAB> <IF> <SPACE> <LPARENTHESIS> <LPARENTHESIS> <MINUS> <LPARENTHESIS> <IDENTIFIER=cy> <PTR> <IDENTIFIER=height> <SPACE> <MULT> <SPACE> <CONSTANT=0.5> <RPARENTHESIS> <RPARENTHESIS> <SPACE> <LESS_THAN> <SPACE> <IDENTIFIER=z> <LBRACKET> <CONSTANT=0> <RBRACKET> <SPACE> <AND> <SPACE> <IDENTIFIER=z> <LBRACKET> <CONSTANT=0> <RBRACKET> <SPACE> <LESS_THAN> <SPACE> <LPARENTHESIS> <LPARENTHESIS> <IDENTIFIER=cy> <PTR> <IDENTIFIER=height> <SPACE> <MULT> <SPACE> <CONSTANT=0.5> <RPARENTHESIS> <RPARENTHESIS> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsExpressionStatement[0m In "ControlStructure" from "Function" line 13":
		<TAB> <TAB> <RETURN> <SPACE> <SEMI_COLON> <NEWLINE>
[36mtest_file_210223.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 14":
		<RBRACE> <NEWLINE>
test_file_210223.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
